apps/sms/test/unit/contacts_test.js

```diff
@@ -467,56 +393,49 @@ suite('Contacts', function(done) {
 
         // navigator.mozContacts.find was called?
         assert.equal(mHistory.length, 1);
-        done();
-      });
+      }).then(done, done);
     });
 
     test('name first, part of email address last', function(done) {
       MockSettings.supportEmailRecipient = true;
       var mozContacts = navigator.mozContacts;
 
-        Contacts.findByString('Pepito a@b', function(contacts) {
-          done(function() {
-          var mHistory = mozContacts.mHistory;
+      Contacts.findByString('Pepito a@b').then(function(contacts) {
+        var mHistory = mozContacts.mHistory;
 
-          // contacts were found
-          assert.ok(Array.isArray(contacts));
-          assert.equal(contacts.length, 1);
+        // contacts were found
+        assert.ok(Array.isArray(contacts));
+        assert.equal(contacts.length, 1);
 
-          // navigator.mozContacts.find was called?
-          assert.equal(mHistory.length, 1);
-        });
-      });
+        // navigator.mozContacts.find was called?
+        assert.equal(mHistory.length, 1);
+      }).then(done, done);
     });
 
     test('part of email address first, name last', function(done) {
       MockSettings.supportEmailRecipient = true;
       var mozContacts = navigator.mozContacts;
 
-      Contacts.findByString('a@b Pepito', function(contacts) {
-        done(function() {
-          var mHistory = mozContacts.mHistory;
+      Contacts.findByString('a@b Pepito').then(function(contacts) {
+        var mHistory = mozContacts.mHistory;
 
-          // contacts were found
-          assert.ok(Array.isArray(contacts));
-          assert.equal(contacts.length, 1);
+        // contacts were found
+        assert.ok(Array.isArray(contacts));
+        assert.equal(contacts.length, 1);
 
-          // navigator.mozContacts.find was called?
-          assert.equal(mHistory.length, 1);
-        });
-      });
+        // navigator.mozContacts.find was called?
+        assert.equal(mHistory.length, 1);
+      }).then(done, done);
     });
 
     test('string search yields a contact without familyName', function(done) {
-      Contacts.findContactByString('julien 123', function(contacts) {
-        done();
-      });
+      Contacts.findContactByString('julien 123').then(() => done(), done);
```

Comment:
actually, it was originally testing that the callback as actually called by in that case. This is not happening if I remove the code https://github.com/mozilla-b2g/gaia/blob/130079ebdb3821275c2e5d5152a70686e2dd133b/apps/sms/js/contacts.js#L60-L62

However with the patch the error is now caught by the "catch" in https://github.com/julienw/gaia/blob/2790a6e448ceacc0b775846548c9c11640b94005/apps/sms/js/contacts.js#L204-L206...

I changed the test to assert that the object is not null, but now I think I should instead remove the "catch" I added and keep a rejected promise. What do you think ?